src/Microsoft.Extensions.Logging.Filter/FilterLoggerSettings.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Filter settings for messages logged by an <see cref="ILogger"/>.
+    /// </summary>
+    public class FilterLoggerSettings : IFilterLoggerSettings, IEnumerable<KeyValuePair<string, LogLevel>>
+    {
+        IChangeToken IFilterLoggerSettings.ChangeToken => null;
+
+        public IDictionary<string, LogLevel> Switches { get; set; } = new Dictionary<string, LogLevel>();
```

Comment:
Well, there's always a "correct" case whether you're case sensitive or not... Plus insensitive comparisons have a long track record of burning us eventually (linux file systems, globalization, etc).

WRT changing later, we've found it's actually easier to go from exact to insensitive in the future than the other way around, because if you go from insensitive to exact folks existing data may no longer match.